Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1598,9 +1598,21 @@ MediaTime MediaPlayerPrivateGStreamer::playbackPosition() const
     // Cached position is marked as non valid here but we might fail to get a new one so initializing to this as "educated guess".
     MediaTime playbackPosition = m_cachedPosition;
 
-    if (GST_CLOCK_TIME_IS_VALID(gstreamerPosition))
+    if (GST_CLOCK_TIME_IS_VALID(gstreamerPosition)) {
         playbackPosition = MediaTime(gstreamerPosition, GST_SECOND);
-    else if (m_canFallBackToLastFinishedSeekPosition)
+        if (isMediaSource() && m_videoDecoderPlatform == GstVideoDecoderPlatform::LibAv && playbackPosition == m_cachedPosition) {
+            auto* playerPrivateMSE = reinterpret_cast<MediaPlayerPrivateGStreamerMSE*>(const_cast<MediaPlayerPrivateGStreamer*>(this));
+            bool isPaused = paused();
+            bool hasFutureTime = playerPrivateMSE->hasFutureTime(m_cachedPosition);
```

Comment:
no need for these 2 local vars

Reply:
They made the code clearer in [MediaPlayerPrivateGStreamerMSE::timeIsProgressing()](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp#L585) the original place where the idea of that code comes from.
Unfortunately, we can't call `timeIsProgressing()` from `currentTime()` because that would cause infinite recursion, so we have to "reimplement" `timeIsProgressing()` inside `playbackPosition()` (which is the inner implementation of `currentTime()`).

But ok, I'll remove them.